### PR TITLE
Fix: Fixed crash that would occur when switching between Network and Recycle Bin

### LINKED ISSUE #10715: App may crash when clicking on Network or Recycle Bin in Sidebar
### Description

This seems to be an issue appearing randomly when using the Sidebar.

I have been able to provoke it only on the Recycle Bin and the Network Drives so far.

### Steps To Reproduce

1. Alternate clicking between items in the Sidebar
2. When clicking on the Recycle Bin or a Network Drive, the application may crash.

### Steps to fix

- Clicking on Network or Recycle Bin should not provoke a crash.

### Files Version

Version: 2.4.1.0 OS Version: 10.0.19045.2251

### Windows Version

W10 Home 22H2 19045.2251

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/10231572/debug.log)

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**
- Closes #10715

**What has been done**
- Wrapped the acquisition of the ShellItem inside a `try...catch...` so the exception raised when a path not valid would be given is no longer making the application crash.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs [hunk 1]
```diff
@@ -85,10 +85,17 @@ public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
 			return Task.FromResult<BaseStorageFile>(null).AsAsyncOperation();
 		}
 
-		private static ShellFileItem GetFile(string path)
+		private static ShellFileItem? GetFile(string path)
```
Comment (yaira2) [inline src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs:88]: Why would this be null?

File: src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs [hunk 2]
```diff
@@ -85,10 +85,17 @@ public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
...
-			using var shellItem = ShellFolderExtensions.GetShellItemFromPathOrPidl(path);
-			return ShellFolderExtensions.GetShellFileItem(shellItem);
+			try
+			{
+				using var shellItem = ShellFolderExtensions.GetShellItemFromPathOrPidl(path);
+				return ShellFolderExtensions.GetShellFileItem(shellItem);
+			}
+			catch
+			{
+				return default;
```
Comment (yaira2) [inline src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs:97]: What circumstances will cause it to catch? Let's add a comment to the code.

File: src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs (hunk 1 above)
Comment (QuaintMako) [reply src/Files.App/Filesystem/StorageItems/ShellStorageFile.cs:88]: The default value of ShellFileItem is null, which could be now raised in the `catch`.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by gave92** [comment]
If it can wait I'd like to take a look at this.

**Comment by QuaintMako** [comment]
@gave92 solution is much cleaner. The issue I offer felt like a makeshift fix, thanks for having looking into it !

**Review by QuaintMako — COMMENTED** [review]
LGTM.

**Comment by gave92** [comment]
Pushed too early, still missing something. You confirm it solves the issue?

**Comment by yaira2** [comment]
> as I found it's needed in some cases.

Do you know where it's still needed? Might be useful to add a comment.

**Comment by gave92** [comment]
> Do you know where it's still needed? Might be useful to add a comment.

Done!

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**